src/uwtools/rocoto.py

```diff
@@ -182,13 +182,60 @@ def _add_task_dependency(self, e: Element, config: dict) -> None:
         :param config: Configuration data for this element.
         """
         e = SubElement(e, STR.dependency)
+        for key, block in config.items():
+            tag, _ = self._tag_name(key)
+            if tag in [STR.taskdep, STR.datadep, STR.timedep]:
+                self._add_task_dependency_operand(e, config={key: block})
+            elif tag in [STR.and_, STR.nand, STR.nor, STR.not_, STR.or_, STR.xor]:
+                self._add_task_dependency_operator(e, config={key: block})
+            elif tag in [STR.streq, STR.strneq]:
+                self._add_task_dependency_strequality(e, config={key: block})
+            else:
+                raise UWConfigError("Unhandled dependency type %s" % tag)
+
+    def _add_task_dependency_operand(self, e: Element, config: dict) -> None:
+        """
+        ???
+        """
         for key, block in config.items():
             tag, _ = self._tag_name(key)
             if tag == STR.taskdep:
                 self._set_attrs(SubElement(e, STR.taskdep), block)
+            elif tag == STR.datadep:
+                self._set_attrs(SubElement(e, STR.datadep), block)
+            else:
+                self._set_attrs(SubElement(e, STR.timedep), block)
+
+    def _add_task_dependency_operator(self, e: Element, config: dict) -> None:
+        """
+        ???
+        """
+        operands = (STR.datadep, STR.taskdep, STR.timedep)
+        operators = (STR.and_, STR.nand, STR.nor, STR.not_, STR.or_, STR.xor)
+        strequality = (STR.streq, STR.strneq)
+
```

Comment:
I'd imagine you'd want to add a subelement to `e` from `config` contents here. For example, if this method gets called from `_add_task_dependency`, `e` will be the top level `dependency` tag. If config contains an operator, we'd need to add that operator as a subelement to `dependency`.

Then, once the subelement is added, we need to go through the `block` to see how the next elements should be handled. So in the next line (217), you'd loop over `block` instead of `config`.